client/lib/importer/actions.js

```diff
@@ -56,7 +57,7 @@ export function cancelImport( siteId, importerId ) {
 		.catch( apiFailure );
 }
 
-export function failUpload( importerId, error ) {
+export function failUpload( importerId, { message: error } ) {
```

Comment:
OK... it pulls out the `message` attribute of the passed in object and passes it as an `error` argument in the function. Still not natural feeling to me, but don't let that hold you up on this.